index.d.ts

```diff
@@ -878,14 +878,11 @@ type ParseTReturn<Key, Res> = Key extends `${infer K1}${_KeySeparator}${infer Re
 
 type TReturnOptionalNull = _ReturnNull extends true ? null : never;
 type TReturnOptionalObjects<TOpt extends TOptions> = _ReturnObjects extends true
-  ? $SpecialObject
+  ? $SpecialObject | string
   : TOpt['returnObjects'] extends true
   ? $SpecialObject
```

Comment:
The same change is missing. But you cannot do it because if you do, the problem is not fixed.